components/domActions.js

```diff
@@ -51,29 +55,137 @@ export const createTodoDomElement = (todo) => {
-    "taskEdit",
-    "icon-edit.svg"
-  );
-  const taskDelete = createButton(
-    "button button-icon",
-    "taskDelete",
-    "icon-delete.svg",
-    () => deleteHandler(id)
-  );
-
-  todoActions.append(taskCompleted, taskEdit, taskDelete);
+  let taskDelete;
+  let taskCompleted;
+  let taskEdit;
+
+  if (isDone) {
+    todoDiv.className = "todo todo-completed";
+
+    taskDelete = createButton(
+      "button button-icon",
+      "taskDelete",
+      "icon-delete.svg",
+      () => deleteHandler(id)
+    );
+
+    todoActions.append(taskDelete);
+    const {
+      months,
+      currentDay,
+      currentMonth,
+      currentYear,
+      minutes,
+      hour,
+      amPm,
+    } = completedAt;
+
+    todoDetails.innerHTML += `<div class="created-at">Completed at: <strong>${hour}:${minutes} ${amPm}, ${currentDay}-${months[currentMonth]}-${currentYear}</strong></div>`;
+  } else {
+    todoDiv.className = "todo";
+
+    taskCompleted = createButton(
+      "button button-icon",
+      "taskCompleted",
+      "icon-tick.svg",
+      () => completeTask(id)
+    );
+
+    taskEdit = createButton(
+      "button button-icon",
+      "taskEdit",
+      "icon-edit.svg",
+      () => editTask(id)
+    );
+
+    taskDelete = createButton(
+      "button button-icon",
+      "taskDelete",
+      "icon-delete.svg",
+      () => deleteHandler(id)
+    );
+
+    todoActions.append(taskCompleted, taskEdit, taskDelete);
+  }
+
+  if (isEditing) {
+    const textArea = document.createElement("textarea");
+    textArea.className = "input-field area";
+    textArea.value = title;
+    editedTitle = title;
+
+    textArea.addEventListener("input", function () {
+      editedTitle = textArea.value;
+    });
+
+    setTimeout(() => {
+      textArea.focus();
+    }, 1);
+
+    todoHeader.innerText = "";
+    todoHeader.append(textArea);
+
+    let updateTaskButton = createButton(
+      "button button-icon",
+      "updateTask",
+      "icon-update.svg",
+      () => updateTask(id)
+    );
+
+    let undoEditButton = createButton(
+      "button button-icon",
+      "taskDelete",
+      "icon-cancel.svg",
+      () => undoEdit(id)
+    );
+
+    todoActions.innerHTML = "";
+    todoActions.append(updateTaskButton, undoEditButton);
+
+    todoDiv.append(todoHeader, todoDetails, todoActions);
+
+    return todoDiv;
+  }
+
   todoDiv.append(todoHeader, todoDetails, todoActions);
 
   return todoDiv;
 };
 
 function deleteHandler(id) {
-  deleteConfirmation("Are you sure you want to delete this?", id);
+  deleteConfirmation("Are you sure?", id);
+}
+
+function updateTask(id) {
+  const editTodo = todos.filter((todo) => todo.id === id);
+  editTodo[0].isEditing = false;
+
+  if (editedTitle.trim() === "") {
+    notifyUser("Todo can not be empty", "error");
+    return;
+  }
+
+  editTodo[0].title = editedTitle.trim();
+
+  setTodos(todos);
+  renderTodoList();
+  notifyUser("Successfully updated todo");
+  localStorage.setItem("todos", JSON.stringify(todos));
+}
```

Comment:
`Array.prototype.find()` is more appropriate here.

```suggestion
function updateTask(id) {
  const editTodo = todos.find(({id: todoId}) => todoId === id);
  
  if(editTodo === undefined){
      return
  }
  
  editTodo.isEditing = false;

  if (editedTitle.trim() === "") {
    notifyUser("Todo can not be empty", "error");
    return;
  }

  editTodo.title = editedTitle.trim();

  setTodos(todos);
  renderTodoList();
  notifyUser("Successfully updated todo");
}
```

Reply:
> if(editTodo === undefined){
      return
  }

Since we are editing and it's an existing item, the value will never return `undefined` right. So, I'm not using this.

Reply:
It's not the point of `find() will never return undefined`, but a point of defensive programming where we should always put guards when a function can return two different values. 

It's a good practice in sense that, it'll make us forcefully use guards in every possible places, whether it's applicable or not.

Also if the project was written in ts, typescript compiler won't even compile this. It would tell us to type narrow or place non-null assertion.